api/lib/models/project/project-data-mapper-impl.ts

```diff
@@ -0,0 +1,236 @@
+      .where('pt.userID = ?', uid);
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts && opts.hackathon ?
+          Promise.resolve(opts.hackathon) :
+          this.activeHackathonDataMapper.activeHackathon
+            .pipe(map(hackathon => hackathon.uid))
+            .toPromise()),
+        );
+    }
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<Project>(query.text, query.values, { cache: true },
+      ))
+      .pipe(map((project: Project[]) => ({ result: 'Success', data: project[0] })),
+      )
+      .toPromise();
+  }
+
+  public async getCount(opts?: IUowOpts): Promise<IDbResult<number>> {
+    let queryBuilder = squel.select({ autoQuoteTableNames: true, autoQuoteFieldNames: false })
+      .from(this.tableName)
+      .field(`COUNT(${this.pkColumnName})`, 'project_count');
+    if (opts && opts.byHackathon) {
+      queryBuilder = queryBuilder
+        .where(
+          'hackathon = ?',
+          await (opts.hackathon ?
+            Promise.resolve(opts.hackathon) :
+            this.activeHackathonDataMapper.activeHackathon
+              .pipe(map(hackathon => hackathon.uid))
+              .toPromise()),
+        );
+    }
+    const query = queryBuilder
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<number>(query, [], { cache: true }),
+    ).pipe(
+      map((result: number[]) => ({ result: 'Success', data: result[0] })),
+    ).toPromise();
+  }
+
+  // need to be checked for conversion
+  public async insert(object: Project): Promise<IDbResult<Project>> {
+    const validation = object.validate();
+    if (!validation.result) {
+      this.logger.warn('Validation failed while adding object.');
+      this.logger.warn(object.dbRepresentation);
+      return Promise.reject({ result: 'error', data: new HttpError(validation.error, 400) });
+    }
+    const query = 'CALL assignTeam (?,?,?,@projectID_out); SELECT @projectID_out as projectID;';
+    const list = [object.project_name, object.team.join(','), object.categories.join(',')];
+    return from(
+      this.sql.query<UidType>(query, list, { cache: false }),
+    ).pipe(
+      map((result: UidType[]) => {
+        object.projectId = result[0];
+        return { result: 'Success', data: object };
+      }),
+    ).toPromise();
+  }
+
+  public update(object: Project): Promise<IDbResult<Project>> {
+    const validation = object.validate();
+    if (!validation.result) {
+      this.logger.warn('Validation failed while adding object.');
+      this.logger.warn(object.dbRepresentation);
+      return Promise.reject({ result: 'error', data: new HttpError(validation.error, 400) });
+    }
+    const query = squel.update({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .table(this.tableName)
+      .setFields(object.dbRepresentation)
```

Comment:
I don't think this will work well. Project contains arrays internally, which sql cannot convert to sensible representations. Instead of using dbRepresentation here, use individual field settings, if the fields in the new object are set. So something like
```
if (object.team) {
  queryBuilder = queryBuilder.set('team', object.team.join(',');
}
```